vnext/Desktop/DevSupportManager.cpp

```diff
@@ -89,21 +95,27 @@ void DevSupportManager::StartPollingLiveReload(const string &debugHost, std::fun
       try {
         Url url(DevServerHelper::get_OnChangeEndpointUrl(debugHost));
         auto const resolveResult = m_resolver.resolve(url.host, url.port);
-        tcp::socket socket{m_context};
-        connect(socket, resolveResult);
 
-        request<string_body> request{verb::get, url.Target(), 11 /*HTTP 1.1*/};
-        request.set(field::host, url.host);
-        request.set(field::user_agent, BOOST_BEAST_VERSION_STRING);
-        request.set(field::connection, "keep-alive");
+        tcp_stream stream{m_context};
+        stream.connect(resolveResult);
+
+        http::request<http::string_body> request{http::verb::get, url.Target(), 11 /*HTTP 1.1*/};
+        request.set(http::field::host, url.host);
+        request.set(http::field::user_agent, BOOST_BEAST_VERSION_STRING);
+        request.set(http::field::connection, "keep-alive");
 
-        write(socket, request);
+        http::write(stream, request);
         flat_buffer buffer;
-        response<dynamic_body> response;
-        read(socket, buffer, response);
+        http::response<http::dynamic_body> response;
+        http::read(stream, buffer, response);
 
         if (response.result_int() == 205 /*ResetContent*/ && !token.is_canceled())
           onChangeCallback();
+
+        error_code ec;
+        stream.socket().shutdown(tcp::socket::shutdown_both, ec);
+        if (ec && ec != errc::not_connected)
+          throw system_error(ec);
       } catch (const std::exception & /*e*/) {
         // Just let the live reload stop working when the connection fails,
```

Comment:
>// J [](start = 8, length = 4)

There should probably be some telemetry, log, or shipassert here

Reply:
Do MSVC or WinSDK have ShipAssert?

Reply:
I don't think there's anything like shipassert built into Windows that would go through the Office telemetry pipeline the same way. I've seen some form of logging in RN, and would be curious to see how we integrate with that.

Reply:
At least add <cassert>'s assert for now. We will need to build up telemetry for GitHub projects at some point.

Reply:
It seems like adding some high level telemetry in general could be super useful if we have an intent to flight out JSI. We probably want more signal than crashes or customer feedback.

Do we have any sort of high level backlog tracking these kinds of things?

Reply:
We have started Open Source Mso project, which could include such tools as ShipAssert and others. This is a rather large but crucial investment. We can discuss ideas during our team meeting.